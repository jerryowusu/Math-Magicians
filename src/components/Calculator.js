import React, { Component } from 'react';
import Button from './Button';
import Display from './Display';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.props = props;
  }

  render() {
    return (
      <div className="calculator">
        <Display />
        <Button />
      </div>
    );
  }
}

export default Calculator;
