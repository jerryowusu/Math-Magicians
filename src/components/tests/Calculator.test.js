import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

it('Calculator component is working fine', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
