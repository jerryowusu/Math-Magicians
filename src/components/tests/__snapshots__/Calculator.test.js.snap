// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component is working fine 1`] = `
<div
  className="calc-body"
>
  <h1
    className="letsdo"
  >
    Let's do some math!
  </h1>
  <div
    className="calculator"
  >
    <div
      className="display"
    >
      <span>
        0
      </span>
      <span />
      <span />
    </div>
    <div
      className="figures"
    >
      <button
        className="clear num"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="ad num"
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="modulo num"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="divide operator"
        name="÷"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
      <button
        className="num"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="num"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="num"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="clear operator"
        name="x"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="num"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="num"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="num"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="clear operator"
        name="-"
        onClick={[Function]}
        type="button"
      >
        –
      </button>
      <button
        className="num"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="num"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="num"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="add operator"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="num"
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="equals operator"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Calculator component is working very well`] = `
    <div
  className="calc-body"
>
  <h1
    className="letsdo"
  >
    Let's do some math!
  </h1>
  <div
    className="calculator"
  >
    <div
      className="display"
    >
      <span>
        0
      </span>
      <span />
      <span />
    </div>
    <div
      className="figures"
    >
      <button
        className="clear num"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="ad num"
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="modulo num"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="divide operator"
        name="÷"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
      <button
        className="num"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="num"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="num"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="clear operator"
        name="x"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="num"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="num"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="num"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="clear operator"
        name="-"
        onClick={[Function]}
        type="button"
      >
        –
      </button>
      <button
        className="num"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="num"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="num"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="add operator"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="num"
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="equals operator"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
